src/js/common/validate.js

```diff
@@ -0,0 +1,343 @@
+/*
+
+A Fluid Component to handle JSON Schema validation.
+
+Pass JSON content to this component's `validate(schemaKey, JSON)` invoker, and it
+will be parsed using [AJV](https://github.com/epoberezkin/ajv).
+
+You are expected to configure one or more schemas which will be read from `options.schemaContents`.  The `schemaKey`
+value passed to `validate` is expected to match one of the keys in `options.schemaContent`.
+
+You can change the behavior of AJV by updating `options.validatorOptions` (check their documentation for the syntax).
+
+If there are no validation errors, we return `undefined`.  If there are validation errors, the raw output generated by
+AJV is transformed using the static function `sanitizeValidationErrors`.
+
+As an example, let's assume that you have a JSON payload like the following that you wish to validate:
+
+  ```
+  {
+    field1: "invalid",
+    category1: {
+      nestedField1: "invalid"
+    }
+  }
+  ```
+
+Assuming that each value was invalid, you might see validation output like the following:
+
+  ```
+  {
+    fieldErrors: {
+      field1: ["should NOT be shorter than 8 characters", "should match pattern \"[A-Z]+\""],
+      category1: {
+        nestedField1: ["nested field is not a valid number."]
+      }
+  }
+  ```
+
+If you have configured your instance to work together with a `gpii.schema.parser`, it will further evolve the error
+messages where possible.
+
+ */
+
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var Ajv = require("ajv");
+
+fluid.registerNamespace("gpii.schema.validator");
```

Comment:
It is all specific to AJV, and it's fair to namespace it that way.  If we have to change again we can talk about abstracting out some kind of general contract at a higher namespace.